### PR TITLE
feat(error.tsx): added error.tsx and not-found.tsx error pages

### LINKED ISSUE #69: [FEAT]: Create a Error Page
**Is your feature request related to a problem? Please describe.**
As we visit any wrong path in website, it shows next.js default error page.
 
image

**Describe the solution you'd like**
We will make our custom error page.

**Describe alternatives you've considered**
NA

**Additional context**
46_404_Error Pages

### PR DESCRIPTION
**Pull request by sub1120**
### Describe your changes

- Created `error.tsx` for any unexpected error.
- Created `not-found.tsx` for 404 page not found error.

### Issue ticket number and link

#69 

### Please confirm the below message,

- [x] I confirm that I have performed a self-review of my code, and written explainable comments in the code, the UI is rendered without any errors, the UI is responsive, dark mode UI is implemented, removed all debug-related code.

### CODE REVIEW

File: src/app/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <div className="space-y-1 md:space-y-2">
+          <h1 className="text-xl font-semibold text-gray-100 dark:text-gray-50 md:text-2xl lg:text-5xl">
+            Error 404
+          </h1>
+          <h2 className="text-lg font-semibold text-gray-900 dark:text-gray-50 md:text-xl lg:text-4xl">
+            Oops! page not found
+          </h2>
+        </div>
+        <p className="md:text-md text-sm text-gray-700 dark:text-gray-50 lg:text-lg">
+          Something went wrong. Its look that your requested could not be found.
```
Comment (shivamvijaywargi) [inline src/app/not-found.tsx]: Requested what could not be found?

File: src/app/not-found.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          It's look like the link is broken or the page is removed.
```
Comment (shivamvijaywargi) [inline src/app/not-found.tsx]: Here and in the above line it should be "It looks like"

File: src/app/error.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            Unexpected Error
+          </h1>
+          <h2 className="text-lg font-semibold text-gray-900 dark:text-gray-50 md:text-xl lg:text-4xl">
+            Oops! Something Went Wrong
+          </h2>
+        </div>
+        <p className="md:text-md text-sm text-gray-700 dark:text-gray-50 lg:text-lg">
+          Its look that some issue occurred while accessing this page. Please
+          retry or contact support{" "}
+          <span className="font-semibold"> dev.lmsgenie@gmail.com</span>
```
Comment (shivamvijaywargi) [inline src/app/error.tsx]: We can put this in an input box with mail to attribute.

File: src/app/error.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        <div className="space-y-1 md:space-y-2">
+          <h1 className="text-xl font-semibold text-gray-100 dark:text-gray-50 md:text-2xl lg:text-5xl">
+            Unexpected Error
+          </h1>
+          <h2 className="text-lg font-semibold text-gray-900 dark:text-gray-50 md:text-xl lg:text-4xl">
+            Oops! Something Went Wrong
+          </h2>
+        </div>
+        <p className="md:text-md text-sm text-gray-700 dark:text-gray-50 lg:text-lg">
+          Its look that some issue occurred while accessing this page. Please
```
Comment (shivamvijaywargi) [inline src/app/error.tsx]: Same here, it should be "It looks that..."

**Review by shivamvijaywargi — CHANGES_REQUESTED** [review]
Please check the comments

File: src/app/not-found.tsx (hunk 1 above)
Comment (sub1120) [reply src/app/not-found.tsx]: Any unknown page /xyz. You can test in your local by giving a wrong path.

File: src/app/not-found.tsx (hunk 2 above)
Comment (sub1120) [reply src/app/not-found.tsx]: Sure

File: src/app/error.tsx (hunk 3 above)
Comment (sub1120) [reply src/app/error.tsx]: Nice idea, you mean like this right?

`<Link href="mailto: dev.lmsgenie@gmail.com">dev.lmsgenie@gmail.com</Link> `

File: src/app/error.tsx (hunk 4 above)
Comment (sub1120) [reply src/app/error.tsx]: Sure

File: src/app/not-found.tsx (hunk 2 above)
Comment (sub1120) [reply src/app/not-found.tsx]: Done.

File: src/app/error.tsx (hunk 3 above)
Comment (sub1120) [reply src/app/error.tsx]: Done. please check

File: src/app/error.tsx (hunk 4 above)
Comment (sub1120) [reply src/app/error.tsx]: Done

File: src/app/not-found.tsx (hunk 1 above)
Comment (shivamvijaywargi) [reply src/app/not-found.tsx]: I was saying it should say something like Your requested route or your request page or your requested resource, etc.., could not be found.

File: src/app/error.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        <div className="space-y-1 md:space-y-2">
+          <h1 className="text-xl font-semibold text-gray-100 dark:text-gray-50 md:text-2xl lg:text-5xl">
+            Unexpected Error
+          </h1>
+          <h2 className="text-lg font-semibold text-gray-900 dark:text-gray-50 md:text-xl lg:text-4xl">
+            Oops! Something Went Wrong
+          </h2>
+        </div>
+        <p className="md:text-md text-sm text-gray-700 dark:text-gray-50 lg:text-lg">
+          It look that some issue occurred while accessing this page. Please
```
Comment (shivamvijaywargi) [inline src/app/error.tsx]: Since you will be reviewing this again, so kindly change it look to it looks

**Review by shivamvijaywargi — CHANGES_REQUESTED** [review]
Please review the comments

File: src/app/not-found.tsx (hunk 1 above)
Comment (sub1120) [reply src/app/not-found.tsx]: Check now, it is fine - `Something went wrong. It looks that your requested page could not be found. It looks like the link is broken or the page is removed.`

File: src/app/error.tsx (hunk 5 above)
Comment (sub1120) [reply src/app/error.tsx]: Done

**Review by shivamvijaywargi — APPROVED** [review]
Good work

**Pull request merged**